scripts/create_newcase

```diff
@@ -1,1003 +1,177 @@
+                        "cimeroot/../")
+
+    # hidden argument indicating called from create_test
+    parser.add_argument("--test", "-test", action="store_true",
+                        help="Used to indicate that create_newcase was called from create_test"
+                        "- do not use otherwise")
+
+    args = parser.parse_args()
+
+    CIME.utils.handle_standard_logging_options(args)
+
+    if args.srcroot is not None:
+        expect(os.path.isdir(args.srcroot),
+               "Input non-default directory srcroot %s does not exist " %args.srcroot)
+        args.srcroot = os.path.abspath(args.srcroot)
+
+    if args.gridfile is not None:
+        expect(os.path.isfile(args.gridfile),
+               "Grid specification file %s does not exist " %args.gridfile)
+
+    if args.pesfile is not None:
+        expect(os.path.isfile(args.pesfile),
+               "Pes specification file %s cannot be found " %args.pesfile)
+
+    if args.user_compset:
+        expect(args.pesfile is not None,
+               "--pesfile is required if --user-compset is set")
+
+    if args.user_grid:
+        expect(args.gridfile is not None,
+               "User grid specification file must be set if the user grid is requested")
+
+    return args.case, args.compset, args.res, args.mach, args.compiler,\
+        args.mpilib, args.project, args.pecount, \
+        args.mach_dir, args.user_mods_dir, args.user_compset, args.pesfile, \
+        args.user_grid, args.gridfile, args.srcroot, args.test
+
+
+###############################################################################
+def _main_func():
+###############################################################################
+
+    case, compset, grid, machine, compiler, \
+        mpilib, project, pecount,  \
+        machine_dir, usermods_dir, user_compset, pesfile, \
+        user_grid, gridfile, srcroot, test \
+        = parse_command_line(sys.argv)
+
+    cimeroot  = os.path.abspath(CIME.utils.get_cime_root())
+
+    if srcroot is None:
+        srcroot  = os.path.dirname(cimeroot)
+
+    # Is case relative or full path?
+    if os.path.isabs(case):
+        caseroot = case
+        case = os.path.basename(case)
+    else:
+        caseroot = os.path.join(os.getcwd(),case)
+
+    # create_test creates the caseroot before calling create_newcase
+    # otherwise throw an error if this directory exists
+    expect(not (os.path.exists(caseroot) and not test),
+           "Case directory %s already exists"%caseroot)
+
+    # Set the case object
+    caseobj = Case(caseroot)
+
+    # Set values for env_case.xml
+    caseobj.set_value("CASE", os.path.basename(case))
+    caseobj.set_value("CASEROOT", caseroot)
+    caseobj.set_value("SRCROOT", srcroot)
+    caseobj.set_value("MACH", machine)
+    if project:
+        caseobj.set_value("PROJECT",project)
+
+    # Configure the Case
+    caseobj.configure(compset, grid, machine,
```

Comment:
Yes, I do think that's where it belongs.   Eventually I would like to see the leftovers from the non-OO like case_setup and batch_maker better integrated into the )) scheme as well.